tm2/pkg/bft/consensus/replay_test.go

```diff
@@ -1172,3 +1172,48 @@ func (ica *initChainApp) InitChain(req abci.RequestInitChain) abci.ResponseInitC
 		Validators: ica.vals,
 	}
 }
+
+func TestHandshakeGenesisResponseDeliverTx(t *testing.T) {
+	t.Parallel()
+
+	app := &initTxsApp{}
+	clientCreator := proxy.NewLocalClientCreator(app)
+
+	config, genesisFile := ResetConfig("handshake_test_")
+	defer os.RemoveAll(config.RootDir)
+	stateDB, state, store := makeStateAndStore(config, genesisFile, "v0.0.0-test")
+
+	// now start the app using the handshake - it should sync
+	genDoc, _ := sm.MakeGenesisDocFromFile(genesisFile)
+	handshaker := NewHandshaker(stateDB, state, store, genDoc)
+	proxyApp := appconn.NewAppConns(clientCreator)
+	if err := proxyApp.Start(); err != nil {
+		t.Fatalf("Error starting proxy app connections: %v", err)
+	}
+	defer proxyApp.Stop()
+	if err := handshaker.Handshake(proxyApp); err != nil {
+		t.Fatalf("Error on abci handshake: %v", err)
+	}
+
+	// check that the genesis transaction results are saved
+	res, err := sm.LoadABCIResponses(stateDB, 0)
+	if err != nil {
+		t.Fatalf("Failed to load genesis ABCI responses: %v", err)
+	}
+
+	if len(res.DeliverTxs) != numInitResponses {
+		t.Fatalf("Expected %d genesis tx responses, got %d", numInitResponses, len(res.DeliverTxs))
+	}
+}
+
+const numInitResponses = 42
+
+type initTxsApp struct {
+	abci.BaseApplication
+}
+
+func (ica *initTxsApp) InitChain(req abci.RequestInitChain) abci.ResponseInitChain {
+	return abci.ResponseInitChain{
+		TxResponses: make([]abci.ResponseDeliverTx, numInitResponses),
+	}
+}
```

Comment:
Please don't define these package-level, but within the test itself 🙏

Reply:
done in 6c412202e2a8be14b3e06e367090640921b8b597